src/storage/storage.cpp

```diff
@@ -331,7 +331,7 @@ void Storage::PopulateLayout(DataLayout &layout)
 
     // load geometries sizes
     {
-        io::FileReader reader(config.geometries_path, io::FileReader::HasNoFingerprint);
+        io::FileReader reader(config.geometries_path, io::FileReader::VerifyFingerprint);
 
         const auto number_of_geometries_indices = reader.ReadVectorSize<unsigned>();
```

Comment:
We need to extract the file size in order to pre-allocate the shared memory block. For this we need to extract the size of every block/vector. 

I couldn't come up with a good abstraction for this yet. But yeah having a dedicated function in `files::` would be good.